### DISCUSSION TITLE
Is there a way to restore the z-order of windows in a layout?

### DISCUSSION THREAD

**Discussion by WillyCornbread** [rxhanson/RectanglePro-Community#216] [Q&A]
I thought that within a layout, if I had 'Bring windows to front' checked, and I sorted my windows in the layout I could control the z-order when the layout is restored. 

This doesn't seem to be the case - I can't figure out what the bring windows to front does exactly, or what the sort of the windows controls? My windows seem to go to random z-orders when restored. I'm on MacOS Ventura.

Would appreciate any help - thanks :)

**Comment by rxhanson** [comment]
Unfortunately, this is looks to be a macOS thing. Rectangle Pro is indeed sending the "bring to front" commands in order, but it likely comes down to asynchronous timing and unpredictable results macOS side. 

If I can think of a way to get around this and get something that works, I'll post back here.

**Reply by WillyCornbread** [reply]
Thanks for the quick reply, it's not a deal-breaker by any means. The software is great.
